### PR TITLE
Implement Heroic Call (summoning of elite knights)

### PR DESCRIPTION
**Pull request by JermyTan**

### CODE REVIEW

File: GameOfRunes/GameOfRunes/GameEngine/GameEngine.swift [hunk 1]
```diff
@@ -247,6 +247,25 @@ class GameEngine {
...
+            return
+        case .darkVortex:
+            gameScene?.deactivateGestureDetection()
+        default:
+            // do nth for other power ups or nil
+            // ensure gesture detection is activated
+            gameScene?.activateGestureDetection()
+        }
+    }
+    /*
```
Comment (dongsiji) [inline GameOfRunes/GameOfRunes/GameEngine/GameEngine.swift]: delete this chunk

**Review by dongsiji — APPROVED** [review]
lgtm

File: GameOfRunes/GameOfRunes/GameEngine/ContactDelegate.swift [hunk 2]
```diff
@@ -19,37 +19,55 @@ class ContactDelegate: NSObject, SKPhysicsContactDelegate {
...
+            default:
+                return
+            }
+        case .playerUnit:
+            guard firstPair.entity.type == .playerUnitEntity,
+                secondPair.entity.type == .endPointEntity,
+                secondPair.collisionType == .enemyEndPoint else {
+                return
+            }
+            gameEngine?.unitReachedLine(firstPair.entity)
```
Comment (dongsiji) [inline GameOfRunes/GameOfRunes/GameEngine/ContactDelegate.swift:70]: would this decrease the player health?

Comment (JermyTan) [reply GameOfRunes/GameOfRunes/GameEngine/ContactDelegate.swift:70]: Wow yes lolol. Good catch on the bug.

**Pull request merged**